lib/pure/strutils.nim

```diff
@@ -906,7 +906,24 @@ proc parseHexInt*(s: string): int {.noSideEffect, procvar,
     of 'A'..'F':
       result = result shl 4 or (ord(s[i]) - ord('A') + 10)
       inc(i)
-    else: raise newException(ValueError, "invalid integer: " & s)
+    else: raise newException(ValueError, "invalid hex integer: " & s)
+
+proc parseBinInt*(s: string): int {.noSideEffect, procvar,
+  rtl, extern: "nsuParseBinInt".} =
+  ## Parses a binary integer value contained in `s`.
+  ##
+  ## If `s` is not a valid binary integer, `ValueError` is raised. `s` can have
+  ## one of the following optional prefixes: ``0b``, ``0B``. Underscores within
+  ## `s` are ignored.
+  var i = 0
+  if i+1 < s.len and s[i] == '0' and (s[i+1] == 'b' or s[i+1] == 'B'): inc(i, 2)
+  while i < s.len:
+    case s[i]
+    of '_': discard
+    of '0'..'1':
+      result = result shl 1 or (ord(s[i]) - ord('0'))
+    else: raise newException(ValueError, "invalid binary integer: " & s)
+    inc(i)
```

Comment:
There is a check missing here testing that we've processed at least one digit.

Reply:
I'm not following. Do you mean regarding the condition to raise the exception?

Reply:
``0b`` or even the empty string would parse successfully with your solution, but shouldn't.

Reply:
Oh, I thought that was intended behavior. `parseHexInt` and `parseOctInt` are working the same way. Should that be changed for all 3 then?

Reply:
> Should that be changed for all 3 then?

I'd say: yes.